applications/serial_lte_modem/doc/FTP_AT_commands.rst

```diff
@@ -130,29 +126,34 @@ Syntax
 
 ::
 
-   AT#XTFTP=<op>,<url>,<port>,<file_path>[,<mode>]
+   AT#XTFTP=<op>,<url>,<port>,<file_path>[,<mode>,<data>]
 
 * The ``<op>`` parameter can accept one of the following values:
 
   * ``1`` - TFTP read request using IP protocol family version 4.
-  * *Currently not supported* ``2`` - TFTP write request using IP protocol family version 4.
+  * ``2`` - TFTP write request using IP protocol family version 4.
   * ``3`` - TFTP read request using IP protocol family version 6.
-  * *Currently not supported* ``4`` - TFTP write request using IP protocol family version 6.
+  * ``4`` - TFTP write request using IP protocol family version 6.
 
 * The ``<url>`` parameter is a string.
   It indicates the hostname or the IP address of the TFTP server.
   Its maximum size is 128 bytes.
   When the parameter is an IP address, it supports both IPv4 and IPv6.
 * The ``<port>`` parameter is an unsigned 16-bit integer (0 - 65535).
-  It represents the TFTP service port on the remote server.
-  Default port 69 is applied if this parameter is omitted or set to ``0``.
+  It represents the TFTP service port on the remote server, which is 69 normally.
 * The ``<file_path>`` parameter is a string.
   It indicates the file path on the TFTP server to read from or write to.
   Its maximum size is 128 bytes.
 * The ``<mode>`` parameter is a string.
   It indicates the three modes defined in TFTP protocol.
   Valid values are ``netascii``, ``octet`` and ``mail``.
   The default value ``octet`` is applied if this parameter is omitted.
+* The ``<data>`` parameter is a string.
+  When ``<op>`` is ``2`` or ``4``, this is the data to be put to remote server.
+
+   .. note::
+      The maximum data size supported in WRITE request is 1024 bytes.
+      Based on protocol, TFTP connection is closed after writing last 512 (or less) bytes.
```

Comment:
```suggestion
      Based on protocol, TFTP connection is closed after writing last 512 (or less) bytes.
```
Can you explain what this means? I'm not sure I quite get it.

Reply:
I'm guessing the termination is the more important part, so putting that in the front might be better:
```suggestion
      The TFTP connection is terminated by writing less than 512 bytes, as defined in the protocol specification (RFC 1350).
```
And if we're referring to the protocol, we should probably link to it as well here, adding it to the `link.txt`.

Reply:
Looks good. The RFC spec is that you can send as large as you like, in 512-byet chunks.
Once the last chunk, 0~511 bytes, is done, the session is over.